src/errors.rs

```diff
@@ -33,28 +33,14 @@ pub enum InternalError {
     InternalInvariantFailed,
     #[error("Paillier error: `{0}`")]
     PaillierError(#[from] paillier::Error),
-    #[error("Failed to convert BigNumber to k256::Scalar, as BigNumber was not in [0,p)")]
-    CouldNotConvertToScalar,
-    #[error("Could not invert a Scalar")]
-    CouldNotInvertScalar,
     #[error("Reached the maximum allowed number of retries")]
     RetryFailed,
     #[error("This Participant was given a message intended for somebody else")]
     WrongMessageRecipient,
     #[error("Encountered a MessageType which was not expected in this context")]
     MisroutedMessage,
-    #[error("Could not construct signature from provided scalars")]
-    SignatureInstantiationError,
-    #[error("Tried to produce a signature without including shares")]
-    NoChainedShares,
     #[error("Storage does not contain the requested item")]
     StorageItemNotFound,
```

Comment:
Not really an error in the same way the others are. Maybe Storage should just return Options?

Reply:
Now that all of our storage is of intermediate states and not of external values, this should be an `InternalInvariantFailed`. It should only happen if we fail to store stuff in our own protocol implementation.

Reply:
Good catch. Here's what I think:

 There are two retrieval functions on `LocalStorage`: `retrieve` and `retrieve_mut`.
 - `retrieve` is called by other functions. A failure in `retrieve` is never conditioned on -- it's propagated up to the caller. From a first pass, it looks like these failures are caused by an internal control flow problem -- like, we called the `gen_round_X_msgs` function but the prerequisites were not actually in place. To me, that means it's an `InternalInvariantFailed`, but it's not guaranteed -- it depends on how it's called. I would add guidance to that function to say, if you need to check whether these things exist, use the `contains` or `contains_for_all_ids` function first, and switch to IIF because 
 - `retrieve_mut` is called by `get_from_storage` only. The caller always checks the error case and returns a default. I think this function should be rewritten to return an `Option` instead of creating and logging an error, since it's not actually an error the way it's used.

Reply:
Since this involves a bit of a refactor, I made #310 with these steps in it.